### PR TITLE
msg/msg_types: let ceph_entity_name::parse() accept string_view

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by badone — APPROVED** [review]
LGTM

**Review by liewegas — APPROVED** [review]

File: src/msg/msg_types.h [hunk 1]
```diff
@@ -81,38 +81,7 @@ class entity_name_t {
...
-      return false;
-    }
-    if (isspace(*start))
-      return false;
-    _num = strtoll(start, end, 10);
-    if (*end == NULL || *end == start)
-      return false;
-    return true;
-  }
+  bool parse(const std::string_view s);
```
Comment (batrick) [inline src/msg/msg_types.h:84]:
```suggestion
  bool parse(std::string_view s);
```

?

Comment (tchaikov) [reply src/msg/msg_types.h]: @batrick updated and repushed.

**Review by batrick — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-04-02_15:50:13-rados-wip-kefu-testing-2021-04-02-1506-distro-basic-gibba/
- https://pulpito.ceph.com/kchai-2021-04-03_11:44:43-rados-wip-kefu-testing-2021-04-03-1318-distro-basic-smithi/

**Pull request merged**